core/services/ocr2/delegate.go

```diff
@@ -351,7 +359,7 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 			OnchainKeyring:         kb,
 		}
 		me := d.monitoringEndpointGen.GenMonitoringEndpoint(spec.ContractID, synchronization.EnhancedEAMercury)
-		return mercury.NewServices(jb, mercuryProvider, d.pipelineRunner, runResults, lggr, oracleArgsNoPlugin, d.cfg, me)
+		return mercury.NewServices(jb, mercuryProvider, d.pipelineRunner, runResults, lggr, oracleArgsNoPlugin, d.cfg, me, chain)
```

Comment:
Using an EVM specific `ChainHeadTracker` type reintroduces the original problem we are trying to avoid here. We need to encapsulate this logic inside of the `mercuryProvider`, and only interact directly with the chain-agnostic API/types from here.

Reply:
Tech debt will be address in https://smartcontract-it.atlassian.net/browse/MERC-619